src/components/Header/Header.jsx

```diff
@@ -1,30 +1,58 @@
 import React, { Component } from 'react';
 import { Link } from 'react-router-dom';
+import { createBrowserHistory } from 'history';
 import { connect } from 'react-redux';
 import queryString from 'query-string';
 import SocialAuth from '../Login/socialAuth';
-import socialAuth, { handleShowAndHide } from '../../actions/socialAuth';
+import socialAuth, {
+  handleShowAndHide,
+  getUserDetails,
+} from '../../actions/socialAuth';
 import defaultAvatar from '../../assets/images/person.png';
 import './Header.scss';
+import { setIsLoggedOut } from '../../actions/logout';
+
+const history = createBrowserHistory();
 
 export class Header extends Component {
-  state = {
-    menu: false,
-  };
+  constructor() {
+    super();
+    this.inputRef = null;
+    this.state = {
+      menu: false,
+      dropdown: false,
+    };
+    this.dropDownRef = null;
+  }
 
   componentDidMount = () => {
     const { location, socialAuth: socialAuthAction } = this.props;
     const url = location.search;
     const userData = queryString.parse(url);
     const { token, user } = userData;
+    const username = localStorage.getItem('username');
+    const localToken = localStorage.getItem('token');
 
     try {
-      if (user) {
+      if (user && !localToken) {
         socialAuthAction(token, JSON.parse(user));
+        localStorage.setItem('username', JSON.parse(user).user.username);
+        history.push('/');
+      } else if (username) {
+        this.props.getUserDetails(username);
       }
     } catch (error) {
       console.log(error);
     }
+
+    window.addEventListener('click', e => {
+      if (
+        e.target.parentNode.className !== 'user-info__avatar' &&
+        this.state.dropdown
+      ) {
+        this.setState({ dropdown: false });
+      }
+    });
```

Comment:
@nkpremices What is the purpose of this block of code? Why not pass the click event directly to the element?

Reply:
This is to close the popup from wherever the user may click on the document @gracelungu